packages/nodes-base/nodes/HttpRequest/v2/HttpRequestV2.node.ts

```diff
@@ -0,0 +1,594 @@
+import path from 'path';
+import { OptionsWithUri } from 'request';
+
+import { IExecuteFunctions } from 'n8n-core';
+import {
+	IBinaryData,
+	IDataObject,
+	ILoadOptionsFunctions,
+	INodeExecutionData,
+	INodePropertyOptions,
+	INodeType,
+	INodeTypeBaseDescription,
+	INodeTypeDescription,
+	NodeApiError,
+	NodeOperationError,
+} from 'n8n-workflow';
+
+import { versionDescription } from './VersionDescription';
+
+interface OptionData {
+	name: string;
+	displayName: string;
+}
+
+interface OptionDataParamters {
+	[key: string]: OptionData;
+}
+
+export class HttpRequestV2 implements INodeType {
+	description: INodeTypeDescription;
+
+	constructor(baseDescription: INodeTypeBaseDescription) {
+		this.description = {
+			...baseDescription,
+			...versionDescription,
+		};
+	}
+
+	methods = {
+		loadOptions: {
+			getNodeCredentialTypes(this: ILoadOptionsFunctions): Promise<INodePropertyOptions[]> {
+				const nodesBaseRoot = path.resolve(__dirname, '..', '..', '..', '..');
+				const packageJson = require(path.resolve(nodesBaseRoot, 'package.json'));
+				const credPaths: string[] = packageJson.n8n.credentials;
+
+				const credOptions = credPaths.reduce<INodePropertyOptions[]>((acc, credPath) => {
+					const match = credPath.match(/(^dist\/credentials\/(?<credType>.*)\.credentials\.js$)/);
+
+					if (!match?.groups) return acc;
+
+					const { credType } = match.groups;
+
+					if (!isSupportedByHttpRequestNode(credType)) return acc;
+
+					const credClass = require(path.resolve(nodesBaseRoot, credPath))[credType];
+
+					return [
+						...acc,
+						{
+							name: new credClass().displayName,
+							value: credType,
+					 	},
+					];
+				}, []);
+
+				return Promise.resolve(credOptions);
+			},
+		},
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+
+		const fullReponseProperties = [
+			'body',
+			'headers',
+			'statusCode',
+			'statusMessage',
+		];
+
+		// TODO: Should have a setting which makes clear that this parameter cannot change for each item
+		const requestMethod = this.getNodeParameter('requestMethod', 0) as string;
+		const parametersAreJson = this.getNodeParameter('jsonParameters', 0) as boolean;
+		const responseFormat = this.getNodeParameter('responseFormat', 0) as string;
+
+		const authenticateWith = this.getNodeParameter('authenticateWith', 0) as 'nodeCredential' | 'genericAuth' | 'none';
```

Comment:
Any specific reason why we can't make all these dynamic? Especially the request method IMO should be something configurable.

Reply:
So I think we can take the opportunity, now that we will be delivering a new version, to also change this behavior. 

I don't think this was 100% intentional and now we cannot really change it since it can break existing workflows, but with a new node version this is something we should consider.

Except for very rare occasions, all fields should be dynamic i.e. parsed for every iteration.